wsmaster/codenvy-hosted-api-resource/src/main/java/com/codenvy/resource/spi/jpa/JpaFreeResourcesLimitDao.java

```diff
@@ -0,0 +1,122 @@
+        try {
+            doStore(resourcesLimit);
+        } catch (RuntimeException e) {
+            throw new ServerException(e.getMessage(), e);
+        }
+    }
+
+    @Override
+    @Transactional
+    public FreeResourcesLimitImpl get(String accountId) throws NotFoundException, ServerException {
+        requireNonNull(accountId, "Required non-null account id");
+        try {
+            return managerProvider.get()
+                                  .createNamedQuery("FreeResourcesLimit.get", FreeResourcesLimitImpl.class)
+                                  .setParameter("accountId", accountId)
+                                  .getSingleResult();
+        } catch (NoResultException e) {
+            throw new NotFoundException("Free resources limit for account '" + accountId + "' was not found");
+        } catch (RuntimeException e) {
+            throw new ServerException(e.getMessage(), e);
+        }
+    }
+
+    @Override
+    @Transactional
+    public Page<FreeResourcesLimitImpl> getAll(int maxItems, int skipCount) throws ServerException {
+        try {
+            final List<FreeResourcesLimitImpl> list = managerProvider.get()
+                                                                     .createNamedQuery("FreeResourcesLimit.getAll",
+                                                                                       FreeResourcesLimitImpl.class)
+                                                                     .setMaxResults(maxItems)
+                                                                     .setFirstResult(skipCount)
+                                                                     .getResultList();
+            return new Page<>(list, skipCount, maxItems, getTotalCount());
+        } catch (RuntimeException e) {
+            throw new ServerException(e.getMessage(), e);
+        }
+    }
+
+    @Override
+    public void remove(String id) throws ServerException {
+        requireNonNull(id, "Required non-null id");
+        try {
+            doRemove(id);
+        } catch (RuntimeException x) {
+            throw new ServerException(x.getLocalizedMessage(), x);
+        }
+    }
+
+    @Transactional
+    protected void doRemove(String id) {
+        final EntityManager manager = managerProvider.get();
+        final FreeResourcesLimitImpl resourcesLimit = manager.find(FreeResourcesLimitImpl.class, id);
+        if (resourcesLimit != null) {
+            manager.remove(resourcesLimit);
+        }
+    }
+
+    @Transactional
+    protected void doStore(FreeResourcesLimitImpl resourcesLimit) throws ServerException {
+        EntityManager manager = managerProvider.get();
+        try {
+            final FreeResourcesLimitImpl existedLimit = get(resourcesLimit.getAccountId());
+            existedLimit.setResources(resourcesLimit.getResources());
+        } catch (NotFoundException n) {
+            manager.persist(resourcesLimit);
+        }
+    }
+
+    @Transactional
```

Comment:
It seems that @Transactional in this case useless because the method with private access modifiers it won't be executed in the transaction.